health/utils.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2018 AMIS Technologies
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package health
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+)
+
+func GRPCServerHealthChecker(addr string) CheckFn {
+	return func(ctx context.Context) error {
+		conn, err := grpc.DialContext(ctx, addr,
+			grpc.WithInsecure(),
+			grpc.WithBlock())
```

Comment:
Should we use a timeout context here to avoid blocking?

Reply:
Yes. I create a timeout context  from parent context to dial grpc server. Fixed in `
2a34bcd`.